frontend/src/js/components/deployments/deployment-wizard/softwaredevices.js

```diff
@@ -45,25 +45,61 @@ const hardCodedStyle = {
   }
 };
 
-export const getDevicesLink = ({ devices, filters = [], group, hasFullFiltering, name }) => {
+export const getDevicesLink = ({ devices, filters = [], group, name }) => {
   let devicesLink = '/devices';
+  if (filters.length) {
+    return `${devicesLink}?${formatDeviceSearch({ pageState: {}, filters, selectedGroup: group })}`;
+  }
+  // older deployments won't have the filter set so we have to try to guess their targets based on other information
   if (devices.length && (!name || isUUID(name))) {
-    devicesLink = `${devicesLink}?id=${devices[0].id}`;
-    if (hasFullFiltering) {
-      devicesLink = `/devices?${devices.map(({ id }) => `id=${id}`).join('&')}`;
-    }
+    devicesLink = `${devicesLink}?${devices.map(({ id }) => `id=${id}`).join('&')}`;
     if (devices.length === 1) {
       const { systemDeviceIds = [] } = devices[0];
       devicesLink = `${devicesLink}${systemDeviceIds.map(id => `&id=${id}`).join('')}`;
     }
-  } else if (group || filters.length) {
+  } else if (group) {
     devicesLink = `${devicesLink}?${formatDeviceSearch({ pageState: {}, filters, selectedGroup: group })}`;
   }
   return devicesLink;
 };
 
+const deploymentFiltersToTargetText = ({ devicesById, filter, idAttribute }) => {
+  const { name, filters = [] } = filter;
+  if (name) {
+    return name;
+  } else if (
```

Comment:
Since `return name`, you could avoid `else if`  making it slightly easier to read IMO.